File: Dec2020/3Dec/DistributeCandies.cpp
/*
Problem name: 575. Distribute Candies
Problem link: https://leetcode.com/problems/distribute-candies
*/

class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        set<int> types;
        for(int i=0;i<candyType.size();++i)
            types.insert(candyType[i]);
        
        if(types.size() <= candyType.size()/2)
            return types.size();
        else
            return candyType.size()/2;
    }
};

File: Dec2020/3Dec/DistributeCandies.java
/*
Problem Name: 575. Distribute Candies
Problem Link: https://leetcode.com/problems/distribute-candies/
*/
class Solution {
    public int distributeCandies(int[] candyType) {
        Set<Integer> set = new HashSet<Integer>();
        for(int i=0;i<candyType.length;i++)
            set.add(candyType[i]);
        return Math.min(candyType.length/2,set.size());
    }
}
